model_utils.py

```diff
@@ -247,18 +259,34 @@ def _retrieve_data(self, raw_dataset):
     def _split_and_retrieve_data(self):
         dataset = self._retrieve_data(self._load_dataset())
 
-        train_set, val_set, test_set = stratified_split(dataset, self.train_proportion, self.types, self.use_test_set)
+        train_set, val_set, test_set, abundances = stratified_split(dataset, self.train_proportion, self.types, self.use_test_set, self.class_balanced)
+
         train_set = train_set.prefetch(tf.data.experimental.AUTOTUNE).cache()
         val_set = val_set.prefetch(tf.data.experimental.AUTOTUNE).cache()
         test_set = test_set.prefetch(tf.data.experimental.AUTOTUNE).cache() if self.use_test_set else None
 
-        if self.has_ids:
+        if self.has_ids: # and self.class_balanced:
             return extract_ids_and_batch(train_set, val_set, test_set, self.batch_size)
+        elif self.has_ids:   
+            train_set = train_set.map(lambda heatmap, label, id: (heatmap, label))
+            val_set = val_set.map(lambda heatmap, label, id: (heatmap, label))
+            train_set = train_set.batch(self.batch_size)
+            val_set = val_set.batch(self.batch_size)
+            if self.use_test_set:
+                test_set = test_set.map(lambda heatmap, label, id: (heatmap, label))
+                test_set = test_set.batch(self.batch_size)
+            return train_set, val_set, test_set
         else:
             train_set = train_set.batch(self.batch_size)
             val_set = val_set.batch(self.batch_size)
             test_set = test_set.batch(self.batch_size)
             return train_set, val_set, test_set
+        
+    def _get_abundances(self):
+        dataset = self._retrieve_data(self._load_dataset())
+
+        _, _, _, abundances = stratified_split(dataset, self.train_proportion, self.types, self.use_test_set, self.class_balanced)
```

Comment:
The difficulty I was having with this is that the last line of `_split_and_retrieve_data` is `return train_set, val_set, test_set` but they are saved in 6 files, `self.train_set, self.val_set, self.test_set, self.train_ids, self.val_ids, self.test_ids`. If I tag `self.abundances` on to the end of that, I get an error because it wasn't the number of expected items to unpack or whatever. Can I set self.abundances in `_split_and_retrieve_data` or will that not carry through the rest of the NN?